### DISCUSSION TITLE
Webhook Service is LIVE

### DISCUSSION THREAD

**Discussion by iklein99** [iklein99/homebridge-smartthings#141] [Announcements]
The Webhook Service is LIVE!

### Why do I need to this?
In order to get updates from your devices, the plugin polls your devices (unless disabled) fairly frequently, but no more than once every 5 seconds.  

If you have a lot of devices, this produces quite a bit of traffic as each device gets polled separately.  This will also result in timeouts occasionally and you may see errors in your logs.

Additionally, the updates are a little delayed.  So if you are using motion detectors, for instance, the delay of 5 seconds in getting an update into Home Kit can be a problem.

This service eliminates the polling.  The plugin will wait for events to come from SmartThings through the Web Hook services.  This results in minimal network traffic and near real-time updates of device state.

### Getting Access
In order to get access, you need to do four main things:

1. Go to [https://www.kleinstudios.net/](https://www.kleinstudios.net/) and purchase a Webhook Access Token.  The service is free for the first 7 days, then you are charged $2.99 per month.  You can cancel any time.
2. Set up a Smart App on your SmartThings account (follow the How To link on the Webhook site)
3. Install the Smart App using the SmartThings mobile app (also described in the How To link on my site)
4. Add the Webhook Access Token to the plugin configuration in Homebridge

While the webhook service has been tested for several weeks now, the purchase site was just launched.  So I'd appreciate any feedback.

Thanks all!

Ira

**Comment by blaine07** [comment]
Kleinstudios URL won’t work?

**Reply by blaine07** [reply]
> Kleinstudios URL won’t work?

Also, in the “Getting Access” portion can you please provide links to the setup instructions for dummy heads like me?

Also, if for some reason, we are to offboard/service is no longer provided could/can things get put back to as is now before subscribing to the service?

**Comment by iklein99** [comment]
Please try again.  Something wrong with the link.  Setup instructions are on the site.  Let me know how it goes.

You can always remove the WebHook Access token from the configuration for the plugin and it will start polling again.  You can cancel the subscription from the site.  Thanks for letting me know about the issue.

**Reply by blaine07** [reply]
Yes, will report back. Mobile this afternoon but this evening I’ll mess with and report back! I’m sure it’ll go well. :-)

**Comment by iklein99** [comment]
All, someone brought to my attention that Samsung moved some things around in their app, and I had to make a change to the directions.  Please note that if you don't see a "Discover" tab at the top of the "Create a Routine" page, you have the latest app, and you can find the smart app config in the "Automations" screen by touching the circle icon at the top right-hand side of the screen.

**Reply by jl0810** [reply]
I don't see any way now to add / discover this app -- has something changed?

**Comment by coryabooth** [comment]


**Reply by iklein99** [reply]
Try and go through the setup screen in the SmartThings app again through to the end.

**Reply by iklein99** [reply]
Corey, I think you may have set the wrong permissions when you set up the Smart App in the developer workspace.  You shouldn't see "control your devices" when you go to do the install.  You should see "See your devices". Check the permissions.  You can always delete the Smart App and create a new one.  The access token you get from my site will work.  See the image below.

<img width="399" alt="image" src="https://user-images.githubusercontent.com/5103250/221383634-c758e01e-f71b-4fb7-b5cd-a9dd9c658229.png">

**Reply by coryabooth** [reply]
> Corey, I think you may have set the wrong permissions when you set up the Smart App in the developer workspace. You shouldn't see "control your devices" when you go to do the install. You should see "See your devices". Check the permissions. You can always delete the Smart App and create a new one. The access token you get from my site will work. See the image below.
> 
> <img alt="image" width="399" src="https://user-images.githubusercontent.com/5103250/221383634-c758e01e-f71b-4fb7-b5cd-a9dd9c658229.png">
![3](https://user-images.githubusercontent.com/5266177/221384464-52f17313-90ed-42ba-b34d-5aa344d7d121.jpg)

**Reply by coryabooth** [reply]
> > Corey, I think you may have set the wrong permissions when you set up the Smart App in the developer workspace. You shouldn't see "control your devices" when you go to do the install. You should see "See your devices". Check the permissions. You can always delete the Smart App and create a new one. The access token you get from my site will work. See the image below.
> > <img alt="image" width="399" src="https://user-images.githubusercontent.com/5103250/221383634-c758e01e-f71b-4fb7-b5cd-a9dd9c658229.png">
> 
> ![3](https://user-images.githubusercontent.com/5266177/221384464-52f17313-90ed-42ba-b34d-5aa344d7d121.jpg)

I fully deleted and recreate the Developer App - still says "Control your devices" and still Error 400

**Reply by coryabooth** [reply]
![image](https://user-images.githubusercontent.com/5266177/221385564-f2eef96a-c946-4b83-b2d2-141e40630667.png)

![image](https://user-images.githubusercontent.com/5266177/221385610-c037dc60-2c56-4568-9269-8966dee71fab.png)

**Comment by joshschools** [comment]
Awesome! This works great, no more errors on my homebridge log. Super responsive! For people having problems accessing the website. I had to put "www" before the website to get the page to load.

**Comment by coryabooth** [comment]
![image](https://user-images.githubusercontent.com/5266177/221385923-6c9f93d7-febc-44ca-871b-d72cd17b830c.png)

**Comment by iklein99** [comment]
That’s expected.  You aren’t making a valid call to the api.  Sent from my iPhoneOn Feb 25, 2023, at 7:32 PM, coryabooth ***@***.***> wrote:﻿

**Comment by gvchagas** [comment]
I have a doubt, for WindFree air conditioners will it be possible to do other tasks, in addition to turning it on and off, how to choose the mode (hot or cold), temperature and ventilation speed?

**Comment by iklein99** [comment]
Air conditioners are not yet supported by the plugin.

**Reply by gvchagas** [reply]
Thanks for the reply! So i'll wait for the support, because i only have a Air conditioner with Smart Things.

**Comment by coryabooth** [comment]
Apologizes for spamming this thread...

I am still stuck however...

I have used CLI to clean everything out, recreated the Webhook in the Developer section numerous time.
I have found the "Test" SmartApp each time, without any issue
I have my Webhook Token
I select "all devices"

But at the final screen, its says "Control" and does not display my selected devices.

And it responds with "Failure: 400 Bad Request"

Is there anything I can check?  Some sort of logging?
Anything to do with Client ID / Secret?
The Developer sites makes heavy mention of "Developer Mode" - perhaps I need adjust a setting in the App for this?

In the Developer section under "Live Log" - all I see is:
CONTROL_EVENT{ "type": "CONTROL_EVENT", "data": "welcome", "lastEventId": "", "origin": "https://spigot-regional.api.smartthings.com" }

In the HomeBridge UI - the "Live Log" says:
[2/25/2023, 11:18:20 PM] [Smartthings Plug (IK)] Could not connect to web hook service: AxiosError: Request failed with status code 401.  Will retry

This thread seems very similar:
https://community.smartthings.com/t/multiple-devices-causes-error-in-smartapp-automation/246749

EDIT

Progress!  Using the thread above, there is a discussion of a potential limit on devices (perhaps 20?)
I reduced my device selection to just my locks (2) - and it worked

I was able to add 18 devices successfully - but it does seem to crash with over 20:
![3](https://user-images.githubusercontent.com/5266177/221394776-5f193a4c-6727-4786-b1a7-3201589f93c3.jpg)

Perhaps this has some confirmation:
https://developer.smartthings.com/docs/advanced/rate-limits/

Guardrail | 30 | Max number of devices per Installed App (Cloud Connected)

**Comment by iklein99** [comment]
Cory do you have any blinds?

Try this.  When you go into the ST app to select your devices, just select a few switches and try it.  Let m know if that works.  It could be a problem with a device type.  I don’t own all of them so something may not be working.  Someone else said they had a problem adding blinds.

**Comment by iklein99** [comment]
Cory, I believe I know your issue.  Hang on while I try something on my end.  Something seems to have gone wrong while registering your account.  The issue seems to be on my side.  Let me have a look.

**Reply by iklein99** [reply]
Cory, it looks like even though you are getting an error, the the devices are installed.  I see 2 installations for you.  One with 33 devices and one with 2.  Here what you should try:

1) configure the plugin by entering your Webhook access token into the plugin config and restart.
2) I assume the second install was just with the 2 devices.  Test that out and see if you are getting the events.  You can see this with Debug on, or change the state of one of those devices and confirm that the state is changing in the Home app
3) If that is the case, try going into the SmartThings app and adding the rest of your devices (maybe a little at a time).  Keep testing, but you should see the events come through.  
4) Let me know after you've added the Access Token and restarted your homebridge so I can see how things look on my end.

**Comment by iklein99** [comment]
Everyone, I believe there may currently be a limit of around 39 devices to be in the subscription.  I'm looking into this and hope to have a fix soon.

**Reply by iklein99** [reply]
Hi all, I found a solution to the problem.  For now, you have to stay around 35 or so devices until I put out a fix.  I promise to get one in soon.  Thanks.

**Reply by iklein99** [reply]
Ok everyone.  Problem is fixed.  When go into the SmartThings app, you no longer have to select devices.  It will monitor all of your devices as long as they support one of the supported capabilities.  Thanks for your patience.

**Comment by joshschools** [comment]
I get this in the Smartthings when I try to approve the access to the
devices.

**Reply by iklein99** [reply]
Josh, not sure if you mean to attach or include something.  All I see is "I get this...".

What are you seeing?

**Reply by joshschools** [reply]
![IMG_0667](https://user-images.githubusercontent.com/16713356/221457100-95de4505-7ef3-435a-9ce3-632ed4e53e65.png)

**Comment by iklein99** [comment]
WhT time zone are you in?Sent from my iPhoneOn Feb 26, 2023, at 9:22 PM, joshschools ***@***.***> wrote:﻿

**Comment by joshschools** [comment]
Central. That was from earlier today. I just tried again. It finally went
through. Not sure what caused it to not work.

**Comment by iklein99** [comment]
I fixed issues around a Samsung limit.  Good news.  Thanks. Sent from my iPhoneOn Feb 26, 2023, at 9:28 PM, joshschools ***@***.***> wrote:﻿
Central. That was from earlier today. I just tried again. It finally went
through. Not sure what caused it to not work.

**Comment by scouts34219** [comment]
I followed the directions provided.  I see my devices in HomeKit and also in Hoobs. I tried to unlock a door lock via HomeKit and it shows unlocked in Hoobs but Smartthings does nothing. Everything I go into the HomeKit Webhook app in the ST app, I get the allow screen but only show sALL DEVICES and when I press Allow I get Something went wrong. Please try to install the SmartApp again. 

![IMG_0BE4B3D7543E-1](https://user-images.githubusercontent.com/23322174/221579236-4632a31d-6eec-435a-aaf7-59d4f8cf1531.jpeg)

**Comment by iklein99** [comment]
Give it about 1/2 hour and try again.  I’m seeing an error from SmartThigns, “too many requests”.  They have a limit.  Not sure how many times you tried.

**Reply by scouts34219** [reply]
Gave it 30 minutes but still get the All devices and not individual devises and when I press Allow it just returns to ST application screen. I tried to unlock via Home and while HOOBS updates ST doesn't do anything. Seems the app is not talking to ST.

I unlocked and locked door in ST and all works but neither HOOBS not HOME was updated.

**Comment by iklein99** [comment]
Not sure I understand.

First, I made a change yesterday so that you don’t select individual devices.  It will now include all devices, but only sends events to the plugin that it cares about.

1. Did you add the Webhook access token to the plugin config?
2. You say that ST doesn’t do anything?  Are you saying that when you try to do something in the Home app, it is not happening on the device?  If so, this has nothing to do with the web hook.  Operations outbound from Home Kit to ST are done via the plugin alone.  If this isn’t working, then you might not have the plug configured properly.  You should remove the Web Hook Access Token from your config (save it off) and restart homebridge.  Outbound commands do not require the web hook.  Make sure it is working.

**Reply by scouts34219** [reply]
OK partly stupid user here :)

I reviewed the README again and saw the config file changed a bit. I updated my config file to what is below via the advance button.

{
    "Name": "Smartthings Plugin",
    "AccessToken": "8f23cca5-XXXXXXXX",
    "WebhookToken: "0071bd30-XXXXXXXX",
    "BaseURL": "https://api.smartthings.com/v1",
    "GarageDoorMaxPoll": 40,
    "PollLocksSeconds": 15,
    "PollDoorsSeconds": 15,
    "PollSensorsSeconds": 5,
    "PollSwitchesAndLightsSeconds": 15
    "platform": "HomeBridgeSmartThings",
    "IgnoreLocations": [
        "My location 1",
        "My location 2"
        ],
    "IgnoreDevices": [
        "Device to ignore 1",
        "Device to ignore 2"
        ]
    }

When I save this it only saves this:

{
    "Name": "Smartthings Plugin",
    "AccessToken": "8f23cca5-xxxxxxxxxx",
    "platform": "HomeBridgeSmartThings"
}

I created a new SmertThings token to use but seem not able to add the WebHook Token to the config. I also uninstalled what I had and reinstalled to send I had the latest clean version :)

**Comment by iklein99** [comment]
Your config file should look something like this.  If you save it, it should not go back to anything like what you showed.  Also, make sure you are running version version 1.5.0 or better of my plugin.  

If you don’t have any devices to ignore, that list should be blank.  When you have a WebhookToken configured, the poll settings are ignored.



{
    "bridge": {
        "name": "Homebridge TEST 3",
        "username": "0E:1C:XX:XX:XX:XX",
        "port": 51019,
        "pin": "031-XX-XXX"
    },
    "accessories": [],
    "platforms": [
        {
            "Name": "Smartthings Plug (IK)",
            "BaseURL": "https://api.smartthings.com/v1/",
            "AccessToken": "XXXXXXXX-fab3-480b-XXXX-23ad45fXXXXX",
            "WebhookToken": "XXXXXXXX-a23e-11ed-XXXX-33c5088XXXXX",
            "GarageDoorMaxPoll": 40,
            "PollLocksSeconds": 10,
            "PollDoorsSeconds": 10,
            "PollSensorsSeconds": 5,
            "PollSwitchesAndLightsSeconds": 10,
            "platform": "HomeBridgeSmartThings",
            "IgnoreLocations": [
                
            ],
            "IgnoreDevices": [
                "Garage Door vSwitch"
            ]
         }
    ]
}

**Reply by scouts34219** [reply]
OK what bridge info are you looking for to be added to the config and do I get that info?

    "bridge": {
        "name": "Homebridge TEST 3",
        "username": "0E:1C:XX:XX:XX:XX",
        "port": 51019,
        "pin": "031-XX-XXX"
    },

I  just used what wa is in your README on how to configure. What you included here saved but still not communication between Homebridge and Smartthings despite seeing the devices in Homebridge. 

Sorry to be a pain but not a programmer just a user :D

**Reply by scouts34219** [reply]
OK think I got it working now, fingers crossed :)  THANK YOU!

**Comment by coryabooth** [comment]
You may want to adjust the language here - specifically the "devices you selected" (and ofcourse toucn)

Finally, you are on a review screen. You can toucn on "See your devices" to see which devices you selected. At the bottom of this screen (scroll down to the bottom) touch "Allow"

**Comment by iklein99** [comment]
Don't mess with your bridge settings.  The only section that needs to be set up is the platform setting for the plugin.  Once you are done.  I may be able to help you more if i see your logs.  See the attached instructions for getting detailed logs and sending them to me.

[Instructions for Producing a Detailed Log.pdf](https://github.com/iklein99/homebridge-smartthings/files/10842785/Instructions.for.Producing.a.Detailed.Log.pdf)

**Comment by psiberfunk** [comment]
If for some reason we prefer to host our own webhook service, is there a way to do that instead of utilizing your service?  While convenient, some would prefer not to have to proxy it through a 3rd party.  I'd be happy to pay a one time fee for the privilege ..

**Reply by iklein99** [reply]
Samsung's SmartThings webhook capability requires access to an endpoint that is a permament https url.  You can't use a typical IP address from a home internet provider that can change and you'd have to acquire a domain name and https certificate.  That is why I deployed it as a service.  Unless you have a permanent IP address to your home,  you'd have to pay anyway for a proxy service.

**Reply by psiberfunk** [reply]
I was thinking something like dynamic DNS or free cloud hosted options, or, for those of us who already own domains and web hosts...

**Reply by whitecat** [reply]
I have a domain name and HTTPS certificate that I can deploy the webhook. Is there someway I can set that up?

**Comment by iklein99** [comment]
All, I found an issue that I fixed where if you had gone through the install smartApp step more than once, you would not get events sent to you.  I fixed that problem.  Please let me know if you have any issues.

**Reply by coryabooth** [reply]
Anything we need to do on our part?
Anything for us to check - perhaps in the logs - that might identify an issue related to this?

**Comment by iklein99** [comment]
No, not now.  I’d have to productize my service.  That’s a big job.  I’d have to charge a lot.

**Comment by clairernovotny** [comment]
Hi, I followed all of the steps, and I'm no longer seeing the lock status change from "unlocking" back to "locked." I hear the lock automatically relocking after 5 seconds, but iOS still shows the animation and different color for the item.

**Reply by iklein99** [reply]
Hi.  Can you send me your user id?  Send it to iklein99@comcast.net?  I'll look in the logs and see if you have everything set up right.

**Reply by clairernovotny** [reply]
Thanks, sent!

**Comment by iklein99** [comment]
Cory,

I can see requests and events flowing back and forth for you.  You should be good.  However I did notice that SmartThings stopped sending events for about 10 minutes or so earlier today.  Not sure why.

Claire us up and running at this point.

**Reply by blaine07** [reply]
> Cory, I can see requests and events flowing back and forth for you. You should be good. However I did notice that SmartThings stopped sending events for about 10 minutes or so earlier today. Not sure why. Claire us up and running at this point.
> […](#)

Maybe this?

![D331F094-CADC-482B-ACBE-DE213F643663](https://user-images.githubusercontent.com/63940698/223228176-08d204b7-843a-4c5a-9f48-d1b297e4cafc.png)

**Comment by iklein99** [comment]
LOL.  Probably.

**Comment by mjhoness** [comment]
Hi,

Bit of a novice at this but really enjoy it's capabilities.
Last night followed the install instructions to a T, easy to follow, no issues.
Thought I might need to leave it for a bit, checked on it this morning, and it is still polling, with error:

[3/11/2023, 7:58:27 AM] [Smartthings Plug (IK)] Could not connect to web hook service: Error: getaddrinfo EAI_AGAIN stwh.kleinstudios.net.  Will retry

Any hints on where I might be going wrong?
If you need more information, please provide a dummy's guide to, I haven't provided any additional information before.

FYI:
username mjhoness
from Australia (incase that has any impact)

Cheers
Matt

**Comment by iklein99** [comment]
That’s an interesting one.  That would indicate a DNS timeout meaning queuing your DNS server, it didn’t get an answer for that URL.  

The plugin will retry like every couple of minutes.  Do you see this error message continuing?  

However, in looking at the server logs, it does appear that your plugin IS connecting.  I can see the requests coming in for you, and I see events being sent back.

Maybe just a moment in time issue.

**Reply by mjhoness** [reply]
Thanks for the quick reply
it is polling so frequently and throwing the following error for multiple devices it is hard to track if it keeps throwing the original error message.

_[3/11/2023, 8:22:44 AM] [homebridge-smartthings-ik] This plugin slows down Homebridge. The read handler for the characteristic 'Status Low Battery' was slow to respond! See https://homebridge.io/w/JtMGR for more info._

My other thought process is that the webhook is working and smartthings no longer being polled and its the way I have homebridge set up

**Comment by iklein99** [comment]
With the web hook setup, it sends one request regardless of how many devices you have, and waits up to 40 seconds for a response.  Then it will send another one, so it isn’t really polling the devices.

If an event occurs, the server send that as soon as it gets it from SmartThings, and then your plugin will handle that event, then make another request to the server.  So there really shouldn’t be much traffic there.

If you remove the web hook token from the config, the plugin will go back to polling each device every few seconds.

When Homebridge wants an update, the plugin requests that from SmartThings and if that takes a while to come back, you will see these errors.  There’s nothing much we can do about that.

Given the error you initially reported and then this, my guess is that you’re getting slow responses through your ISP.

I’d keep an eye on it, and if it continues, let me know.  I’ll send you instructions on how to capture details logs that I can look at.

**Comment by mjhoness** [comment]
Thanks for the education!
You should add it to your site or the smartthings repository in github - for non coders like myself it helped me heaps!
Why? Long story short, I have been battling my ISP for a while now due to poor performance with frequent drop outs. Within a week I should have a service upgrade capable of delivering >250mps download speeds from my current ~30! ie I will have fibre to the property!
So once that is done I'll revisit the logs to see if their is significant performance improvements and let you know!
Your quick education has helped heaps!

**Comment by iklein99** [comment]
It appears that a small number folks that have set up the webhook smart app are not receiving events from smartthings.  This is because somehow the ID that gets created when you install the app isn't matching what is in the database.  I'm not sure how this happened - I'm looking at it, but for now, if you are experiencing issues, perform the following:

1. Open the ST app on your mobile device
2. Go to the Automations tab
3. Scroll down to the Smart App for the webhook and open that
4. Click 'Done' at the bottom, then 'Allow' on the next screen
5. Wait about 5 minutes and see if your getting updates

Sorry about the issue.

**Reply by blaine07** [reply]
> It appears that a small number folks that have set up the webhook smart app are not receiving events from smartthings. This is because somehow the ID that gets created when you install the app isn't matching what is in the database. I'm not sure how this happened - I'm looking at it, but for now, if you are experiencing issues, perform the following:
> 
> 1. Open the ST app on your mobile device
> 2. Go to the Automations tab
> 3. Scroll down to the Smart App for the webhook and open that
> 4. Click 'Done' at the bottom, then 'Allow' on the next screen
> 5. Wait about 5 minutes and see if your getting updates
> 
> Sorry about the issue.

Hey, you, yes YOU— thank YOU for keeping this going and working for US. You’re doing a great job at solving issues as they come around and WE APPRECIATE IT! 😀

**Comment by swav8tor** [comment]
Just signed up and installed.  So far working as advertised.  Thank You.

**Comment by alaw168** [comment]
Signed up and installed over the weekend. Works great so far.

A couple of minor things.

1) I'm still getting sporadic messages in the log like this:

[30/05/2023, 09:18:52] [homebridge-smartthings-ik] This plugin slows down Homebridge. The read handler for the characteristic 'Battery Level' was slow to respond! See https://homebridge.io/w/JtMGR for more info.

Does it mean it is still polling from ST?

2) Somewhere in the log that mentions webhook is in use would be nice. Also, regular log messages on events (I can see them in debug logs) would also be nice.

**Comment by jeffp94114** [comment]
Hmmm, I got all the way through the instructions ok, but at the very end when I restart Homebridge I get, "No plugin was found for the platform "HomeBridgeSmartThings" in your config.json. Please make sure the corresponding plugin is installed correctly."

Is there maybe an alternative name I should be trying there?  Thx

**Reply by iklein99** [reply]
@jeffp94114 , Can you go into the plugins screen and send a screenshot of the plugins page? Also send me your config file, remove your access tokens first.

**Reply by jeffp94114** [reply]
Ha, my bad - I did everything except remember to install the plug-in itself in Homebridge :-)  Added it in and now everything's working, thanks!

**Comment by iklein99** [comment]
Can you go into the plugins screen and send a screenshot of the plugins page?  Also send me your config file, remove your access tokens first.

**Comment by iklein99** [comment]
No.   Sorry.

**Comment by nicofriedrich** [comment]
Hey @iklein99, don't know what I'm doing wrong here:

Did the whole setup. 
- ST dev portal
- ST app
- HB config

But I can't see my devices. So no webhook is being called on an update.
Just the welcome msg in `ST dev portal`:

```json
{ "type": "CONTROL_EVENT", "data": "welcome", "lastEventId": "", "origin": "https://spigot-regional.api.smartthings.com" }
```
Do you have any tips?

I saw that there might be a problem with too much devices but I just have 6 at the moment.

**Reply by iklein99** [reply]
@nicofriedrich , you say you don't see any devices?  You mean you don't see your devices in the Home app?

**Comment by nicofriedrich** [comment]
> you say you don't see any devices?

Sry, I don't see devices on ST-APP
Let me share a screen:
![IMG_3A8BE38F6544-1](https://github.com/iklein99/homebridge-smartthings/assets/43110822/a796a9d0-3689-41c8-9efd-6f8df90f5247)

**Reply by iklein99** [reply]
What is the english translation of that drop down selection?  It should say "see your devices".  I am interested what devices you see in the Apple Home app.  You don't need the webhook to see devices.

**Comment by nicofriedrich** [comment]
> What is the english translation of that drop down selection? It should say "see your devices". I am interested what devices you see in the Apple Home app. You don't need the webhook to see devices.

Sry its in german, but the list of devices is empty when I try to allow specific devices.  (last step from webhook app)
But they are all (6 devices) in ST-App and also in your Homebridge-Plugin. 

English-translation:
Homebridge Webhook wants at current location:
Display all devices: (drop down is active)
<empty list>
All devices

ActionButtons:
Deny Allow

**Comment by sean8151** [comment]
Hi @iklein99 ,

I hope you are doing well. I followed all the instruction, but the last step "configure the plugin" in my page is different, there is no "WebhookToken" line, I only see the AccessToken line, can you please help me with this? Thank you!

FYI: I'm using Hoobs, and the version of Smartthings Ik is 1.5.20.

Best,
Aaron

**Reply by sean8151** [reply]
I use plugin "Configuration", "Advanced" to added a line with "WebhookToken": "……", under AccessToken, seems does not working.

**Reply by iklein99** [reply]
You can edit the config file directly.  You can do it in the UI or edit it directly in the .homebridge/config.json file.  It should look what the link below.

<img width="519" alt="Screenshot 2023-11-20 at 11 00 43 AM" src="https://github.com/iklein99/homebridge-smartthings/assets/5103250/cc832cd7-1312-4994-88eb-11ff66d694fb">

**Reply by sean8151** [reply]
I did. How can I test if the webhook is working or not? Thank you!

<img width="481" alt="Screenshot 2023-11-20 at 8 20 54 AM" src="https://github.com/iklein99/homebridge-smartthings/assets/133162033/04ae77b6-4fc6-4fac-8f03-b94d46448192">

**Reply by iklein99** [reply]
I can confirm you are working.  I looked in my logs and see activity:

11/20/2023, 4:25:21 PM: info: CLIENT REQUEST for aaron8151
11/20/2023, 4:26:46 PM: info: CLIENT REQUEST for aaron8151
11/20/2023, 4:27:22 PM: info: Event(s) detected for user: aaron8151
11/20/2023, 4:27:23 PM: info: CLIENT REQUEST for aaron8151
11/20/2023, 4:28:38 PM: info: Event(s) detected for user: aaron8151
11/20/2023, 4:28:39 PM: info: CLIENT REQUEST for aaron8151
11/20/2023, 4:29:36 PM: info: Event(s) detected for user: aaron8151
11/20/2023, 4:29:36 PM: info: CLIENT REQUEST for aaron8151
11/20/2023, 4:30:10 PM: info: CLIENT REQUEST for aaron8151
11/20/2023, 4:31:17 PM: info: CLIENT REQUEST for aaron8151

Note that the times are UTC.

**Reply by sean8151** [reply]
I see, however, that from the Home app, the devices still need about 25 seconds or so to change their status from 'updating'. Do you have any idea why the devices status changes so slowly?

**Reply by sean8151** [reply]
![IMG_1047](https://github.com/iklein99/homebridge-smartthings/assets/133162033/86f5bbc5-b49f-483b-b2fc-8ef7dc0e9456)

**Reply by iklein99** [reply]
That has nothing to do with the webhook.  It is either your Hoobs homebridge running slow or your home hub.  If you follow the directions, I will take a look at the timing in your logs.

[Instructions for Producing a Detailed Log.pdf](https://github.com/iklein99/homebridge-smartthings/files/13417397/Instructions.for.Producing.a.Detailed.Log.pdf)

**Reply by sean8151** [reply]
I found the problem! I deleted the one device that was offline. All other devices change status right away! THANK YOU!

**Comment by iklein99** [comment]
Look at the logs.  Do you see errors?  If not, then you should be able to see by physically changing a device state and see if that change is reflected in the Apple Home app.

**Reply by nicofriedrich** [reply]
Can u do the same for me? Just see if it is working on your end?
What do u need from me? UserName: `nictc`

**Reply by iklein99** [reply]
I see the client requests coming from your homebridge, but no events coming from Smartthings.

**Reply by nicofriedrich** [reply]
OK I guess I will restart from scratch once more... thx

**Reply by iklein99** [reply]
YOu need to make sure that the steps to set up the smart app in Smartthings completed properly.

**Comment by nicofriedrich-ibm** [comment]
So after restart from scratch: Following all steps, I cannot get it to run:
Account: `nictc`
Looks like smarthtings app + webhook cannot access my devices.
Location: `My Home`
Count of devices: `7`
I can't see any errors in `HomeBridge`.
Can you help me how to fix it? 

What is odd to me is when I click on `See all devices` I don't see a single one see attached screenshot from your HowTow.
![app_setup_6 c0e357c3aca7179bfe4b](https://github.com/iklein99/homebridge-smartthings/assets/151632204/9e51369b-6ec5-48ce-b787-f9310a15a3f3)

**Reply by coryabooth** [reply]
Do you have any pihole or some other traffic controller running?

**Reply by nicofriedrich-ibm** [reply]
no I don't.

**Reply by nicofriedrich-ibm** [reply]
<img width="907" alt="Bildschirmfoto 2023-12-11 um 08 33 14" src="https://github.com/iklein99/homebridge-smartthings/assets/151632204/22f981cc-b041-4a04-a7b5-a896d251405c">
This is how webhook app looks like in Samsung: SM ^^

**Reply by sebtobar** [reply]
@nicofriedrich-ibm did you get a solution to this?

@iklein99 I seem to be having the same issue. I’ve set everything up according to the instructions but I can’t see my device (I only have 1, 2 if you count a legacy one that isn’t set up on my ST app)

The device is a robot vacuum, not sure if it’s to do with that.

**Comment by montag2k** [comment]
Hey Ira!

Thank you for providing this API. I’m signed up (as gogobonobo). I was hoping this would solve the problem I’m having of button events not making it to Homebridge/Homekit, but it looks like even though button presses are showing up in SmartThings, they aren’t making it across. 

Is there any way you could check to see if it’s making it over?

**Reply by montag2k** [reply]
Well I don’t know what I did but it’s working now. Huh.

**Reply by iklein99** [reply]
Make sure you have a "Home Hub" set up which is separate from Homebridge.  I have on running on an old ipad, that I have to restart occasionally.   

https://support.apple.com/en-us/HT207057

**Reply by montag2k** [reply]
Yup I’ve got an AppleTV doing the job. I’m not exactly sure what the problem was but turning on debug mode showed the button events making it to Homebridge, and then all of a sudden it started triggering the event I set up in the Home app.

**Comment by coryabooth** [comment]
I advise anyone using an iPad as a HomeHub to move to HomePod or Apple TV as the primary hub.

https://9to5mac.com/2022/06/28/apple-says-ipad-can-still-be-a-home-hub-in-ios-16-as-long-as-the-new-homekit-features-dont-matter-to-you/

https://9to5mac.com/2022/06/08/ios-16-homekit-hubs/

**Comment by jjeaton** [comment]
Any users that have seen a big improvement with the webhook service, could you let me know? I regularly have issues with ST where it thinks everything is working but flipping switches doesn't do anything, until I restart the child bridge. No errors in the log or anything. I only have 2 ST devices, z-wave switch and lock. Just curious if anyone has seen a huge improvement with the webhook

**Comment by mathblad** [comment]
Hello, I needed to change my payment method for your web hook and in order to do that I need to cancel my subscription and resubscribe. In doing so I lost basically a whole month of your service. Can you help me?

**Comment by kshartman** [comment]
@iklein99 have you stopped running the web service.  I get errors when trying to connect.

**Reply by kshartman** [reply]
<img width="1309" height="43" alt="image" src="https://github.com/user-attachments/assets/209805f5-45f8-4e1b-ab32-b232445a1693" />

**Comment by iklein99** [comment]
Looks like there is an issue causing slowness. I’m working on it.  Sent from my iPhoneOn Jul 18, 2025, at 1:53 PM, Shane Hartman ***@***.***> wrote:﻿
@iklein99 have you stopped running the web service.  I get errors when trying to connect.

**Comment by iklein99** [comment]
Errors should have gone away now.  Had to restart the server.  Thanks.

**Reply by pauldonahue** [reply]
Hello. First, thank you very much for all of your work on creating this plugin and offering the webhook service. Having over 80 devices in SmartThings bringing them into HomeKit via your plugin is delightful!

Nextly, I'm still seeeing lots of connection errors as of 15:00 EDT July 23, 2025.
<img width="901" height="161" alt="image" src="https://github.com/user-attachments/assets/b430ee6b-f4ea-48e2-b7b5-ee976d189475" />

**Comment by iklein99** [comment]
Ok.  I just restarted the server.  That should clear it up.  Not sure why this happened again.

**Reply by pauldonahue** [reply]
Thank you!

**Comment by MRedRaider81** [comment]
@iklein99  looks like there may be another server side issue:
<img width="1633" height="49" alt="image" src="https://github.com/user-attachments/assets/327cf478-10a8-4443-81f8-c0837cf2ea5f" />

Thanks for all you do!
